### PR TITLE
Match community names case-insensitively

### PR DESCRIPTION
**Pull request by jstvz**
This PR:
- Updates the community exists error message to match the new ExperienceBundle message.
- Matches community names case-insensitively.

Fixes an issue discovered by @Julian88Tex in SalesforceFoundation/SAL#2116.

# Critical Changes

# Changes

# Issues Closed
- Fixed an issue where the `create_community` task would throw an exception on an existing community despite `skip_existing` being true.

### CODE REVIEW

File: cumulusci/tasks/salesforce/CreateCommunity.py [hunk 1]
```diff
@@ -119,5 +120,5 @@ def _poll_action(self):
 
     def _get_community(self):
         community_list = self.sf.restful("connect/communities")["communities"]
-        communities = {c["name"]: c for c in community_list}
-        return communities.get(self.options["name"])
+        communities = {c["name"].casefold(): c for c in community_list}
```
Comment (prescod) [inline cumulusci/tasks/salesforce/CreateCommunity.py:123]: We use .lower() everywhere else. Is there anything special about this context?

Comment (jstvz) [reply cumulusci/tasks/salesforce/CreateCommunity.py:123]: Nothing special here. I just stumbled on it in the docs and thought "huh, better caseless matching of unicode strings" and added it here.

Comment (jstvz) [reply cumulusci/tasks/salesforce/CreateCommunity.py:123]: Following up, @prescod. Are you requesting a change here?

**Review by prescod — DISMISSED** [review]
Sorry my comment was pending. I can approve but I'd rather not switch to casefold pervasively unless we are confident that its better (i.e. more like Salesforce).

**Comment by jstvz** [comment]
> Sorry my comment was pending. 

Weird. It actually put the comments in created order, but out of chronological order as displayed.

> I can approve but I'd rather not switch to casefold pervasively unless we are confident that its better (i.e. more like Salesforce).

You're right. I looked it up and this particular resource is calling [`String.equalsIgnoreCase()`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html#equalsIgnoreCase(java.lang.String)). 

Updated.

File: cumulusci/tasks/salesforce/CreateCommunity.py [hunk 2]
```diff
@@ -119,5 +120,5 @@ def _poll_action(self):
 
     def _get_community(self):
         community_list = self.sf.restful("connect/communities")["communities"]
-        communities = {c["name"]: c for c in community_list}
-        return communities.get(self.options["name"])
+        communities = {c["name"].lower().upper(): c for c in community_list}
```
Comment (Br4nd0R) [inline cumulusci/tasks/salesforce/CreateCommunity.py:123]: Was going to ask why this needed but then read the conversation thread. TIL.

**Review by Br4nd0R — APPROVED** [review]
Nice work. All looks good.

File: cumulusci/tasks/salesforce/CreateCommunity.py (hunk 2 above)
Comment (jstvz) [reply cumulusci/tasks/salesforce/CreateCommunity.py:123]: Yeah, I went all the way down the rabbit hole on this one.

**Pull request merged**